src/bkl/interpreter/builder.py

```diff
@@ -221,10 +221,23 @@ def on_assignment(self, node):
             if previous_value is None:
                 # appending to inheritable list property with empty default
                 value = ListExpr(new_values)
-            elif isinstance(previous_value.value, ListExpr):
-                value = ListExpr(previous_value.value.items + new_values)
             else:
-                value = ListExpr([previous_value.value] + new_values)
+                if isinstance(previous_value.value, ListExpr):
+                    previous_values = previous_value.value.items
+                else:
+                    previous_values = [previous_value.value]
+
+                combined_values = previous_values
+                if var.idempotent:
+                    # avoid adding the items more than once
+                    for new_value in new_values:
+                        if not new_value in combined_values:
```

Comment:
Argh, actually this doesn't always work because comparing two `Expr` objects may throw if one of them is conditional.

I could probably hack around it somehow, but I wonder if this doesn't mean that the whole approach is flawed and that I need to postpone pruning the duplicate dependencies until all the conditions can be actually evaluated (but when and where will this be the case?).